spark/v3.3/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestDelete.java

```diff
@@ -764,12 +765,18 @@ public synchronized void testDeleteWithSerializableIsolation() throws Interrupte
     Future<?> appendFuture =
         executorService.submit(
             () -> {
+              SparkSession sparkSession = spark.cloneSession();
               for (int numOperations = 0; numOperations < Integer.MAX_VALUE; numOperations++) {
                 while (barrier.get() < numOperations * 2) {
                   sleep(10);
                 }
 
                 try {
+                  Dataset<Row> inputDF =
```

Comment:
I was running into some weird thread-local issues without cloning the session in the thread.